apps/full-site-editing/full-site-editing-plugin/editor-gutenboarding-launch/index.ts

```diff
@@ -53,6 +53,42 @@ function updateEditor() {
 		const isMobile = window.innerWidth < 768;
 		const isNewLaunch = window?.calypsoifyGutenberg?.isNewLaunch;
 
+		// Assert reason: We have an early return above with optional and falsy values. This should be a string.
+		const launchHref = window?.calypsoifyGutenberg?.frankenflowUrl as string;
+
+		// On mobile there is not enough space to display "Complete setup" label.
+		const launchLabel = isMobile
+			? __( 'Launch', 'full-site-editing' )
+			: __( 'Complete setup', 'full-site-editing' );
+
+		const saveAndNavigate = async () => {
+			await dispatch( 'core/editor' ).savePost();
+			// Using window.top to escape from the editor iframe on WordPress.com
+			window.top.location.href = launchHref;
+		};
+
+		const handleLaunch = ( e: Event ) => {
+			// Disable href navigation
+			e.preventDefault();
+
+			const shouldOpenNewFlow = isNewLaunch && ! isMobile;
+
+			recordTracksEvent( 'calypso_newsite_editor_launch_click', {
```

Comment:
`launch_click` could be `complete_setup` click? `launch_click` we might want to reserve for the last step in that list.